pkg/cloudprovider/backup_service.go

```diff
@@ -45,8 +45,11 @@ type BackupService interface {
 	// downloading or reading the file from the cloud API.
 	DownloadBackup(bucket, name string) (io.ReadCloser, error)
 
-	// DeleteBackup deletes the backup content in object storage for the given api.Backup.
-	DeleteBackup(bucket, backupName string) error
+	// DeleteBackup deletes the backup content in object storage for the given backup.
+	DeleteBackupFile(bucket, backupName string) error
+
+	// DeleteBackup deletes the backup metadata file in object storage for the given backup.
+	DeleteBackupMetadataFile(bucket, backupName string) error
```

Comment:
Heads up - in my backup-logs PR, I'm going to make it so that the way we delete a backup is to list everything matching the backup prefix (e.g. list bucket/my-backup/*) and then delete each item. We'll need this as log files and anything else gets added into the backup dir. Do you want to merge this as-is and then I'll replace it in my PR?

Reply:
yeah, that's fine.

Reply:
Do you want me to delete everything other than ark-backup.json first, and if all the deletes succeed, delete that last?